File: repaso1/repaso1_c/src/Contacto.cpp
#include "Contacto.h"

Contacto::Contacto(char* nom, int num, char* mail): telefono(num)
{
    nombre=new char[strlen(nom)+1];
    strcpy(nombre, nom);

    email=new char[strlen(mail)+1];
    strcpy(email,mail);

}

Contacto::Contacto(const Contacto &c){
    nombre=new char[strlen(c.nombre)+1];
    strcpy(nombre, c.nombre);
    telefono=c.telefono;
    email=new char[strlen(c.email)+1];
    strcpy(email, c.email);

}

Contacto::~Contacto()
{
    delete[]nombre;
    delete[]email;
}

Contacto& Contacto::operator=(const Contacto &c){
    if(this!=c){
        delete[]nombre;
        delete[]email;
        nombre=new char[strlen(c.nombre)+1];
        strcpy(nombre,c.nombre);

        telefono=c.telefono;
        email=new char[strlen(c.email)+1];
        strcpy(email, c.email);
    }
    return *this;
}

ostream& operator<<(ostream &s, const Contacto &c){
    s << c.getNombre() << " (" << c.getTelefono() << ")-"<<c.getEmail();
    return s;
}

File: repaso1/repaso1_java/src/repaso1_java/Contacto.java
package repaso1_java;

public class Contacto implements Cloneable {
    private String nombre;
    private int telefono;
    private String email;
    
    public Contacto(String nom, int tel, String mail) {
        nombre = nom;
        telefono = tel;
        email = mail;
    }
    
    public Contacto(Contacto c) {
        nombre = c.nombre;
        telefono = c.telefono;
        email = c.email;
    }
    
    public String getNombre() { return nombre; }
    public int getTelefono() { return telefono; }
    public String getEmail() { return email; }
    
    public Object clone() {
        return new Contacto(this);
    }
    
    public boolean equals(Object o) {
        if (this.getClass() != o.getClass()) return false;
        Contacto c = (Contacto) o;
        return nombre.equals(c.nombre) && telefono == c.telefono && 
               email.equals(c.email);
    }
    
    public String toString() {
        return nombre + " (" + telefono + ") - " + email;
    }
}
